core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataService.js

```diff
@@ -153,32 +127,17 @@ oppia.factory('TrainingDataService', [
         _removeAnswer(answer);
 
         var answerGroup = ResponsesService.getAnswerGroup(answerGroupIndex);
-        var rules = answerGroup.rules;
-
-        // Ensure the answer group has a classifier rule.
-        var classifierRule = null;
-        for (var i = 0; i < rules.length; i++) {
-          var rule = rules[i];
-          if (rule.type === RULE_TYPE_CLASSIFIER) {
-            classifierRule = rule;
-            break;
-          }
-        }
-        if (!classifierRule) {
-          // Create new classifier rule for classification.
-          classifierRule = RuleObjectFactory.createNewClassifierRule();
-          rules.push(classifierRule);
-        }
+        var trainingData = answerGroup.training_data;
 
         // Train the rule to include this answer, but only if it's not already
         // in the training data.
         if (_getIndexOfTrainingData(
```

Comment:
Is the index actually important? Might want to have a _doesTrainingDataContain(answer, trainingData) instead.

Reply:
Yes index is required because we have to remove it from previous class and add it to new class. However this is related to training data interface and don't worry about it as this is going to be created from scratch in ML GSoC Project.

Reply:
Oh ok. But I think you want to still declare a second private function anyway that just checks membership (it can rely on _getIndexOfTrainingData), since this is what this part of the code really cares about (and I think that will be a common operation anyway).

Also can we use the term "trainingData" just for the aggregate of all answers, and "trainedAnswer" for a single trained answer? Using it for both is a bit confusing.

Reply:
OK, you shouldn't focus on this service. Last time I checked it was broken. I don't know when or how but it didn't work and we already have the project to recreate them from scratch.

Reply:
OK.